### PR TITLE
JAMES-3138 Reactify CurrentQuotaManager

### PR DESCRIPTION
**Pull request by Arsnael**
Based on #3275 

Regarding this comment: https://github.com/linagora/james-project/pull/3284#discussion_r408710405

### CODE REVIEW

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ public void event(Event event) throws MailboxException {
 
     private void handleExpungedEvent(Expunged expunged, QuotaRoot quotaRoot) {
         computeQuotaOperation(expunged, quotaRoot).ifPresent(Throwing.<QuotaOperation>consumer(quotaOperation -> {
-            currentQuotaManager.decrease(quotaOperation);
+            currentQuotaManager.decrease(quotaOperation).block();
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java:92]: We sould avoid a double block.

A.then(B).block will prevent this.

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java [hunk 2]
```diff
@@ -107,7 +107,7 @@ private void handleExpungedEvent(Expunged expunged, QuotaRoot quotaRoot) {
 
     private void handleAddedEvent(Added added, QuotaRoot quotaRoot) {
         computeQuotaOperation(added, quotaRoot).ifPresent(Throwing.<QuotaOperation>consumer(quotaOperation -> {
-            currentQuotaManager.increase(quotaOperation);
+            currentQuotaManager.increase(quotaOperation).block();
```
Comment (chibenwa) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java:110]: We sould avoid a double block. (eventBus call below)

A.then(B).block will prevent this.

File: mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/quota/InMemoryCurrentQuotaManager.java [hunk 3]
```diff
@@ -55,38 +56,34 @@ public InMemoryCurrentQuotaManager(CurrentQuotaCalculator quotaCalculator, Sessi
...
+            .then();
     }
 
-    private void updateQuota(QuotaRoot quotaRoot, UnaryOperator<CurrentQuotas> quotaFunction) throws MailboxException {
-        try {
-            quotaCache.get(quotaRoot).updateAndGet(quotaFunction);
-        } catch (ExecutionException e) {
-            throw new MailboxException("Exception caught", e);
-        }
+    private Throwable propagateException(Throwable throwable) {
```
Comment (mbaechler) [inline mailbox/memory/src/main/java/org/apache/james/mailbox/inmemory/quota/InMemoryCurrentQuotaManager.java:86]: wrapAsMailboxException?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

(build result lost)

**Comment by chibenwa** [comment]
```
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] [ERROR] Failures: 
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] [ERROR]   MailboxManagerTest$EventTests.addingMessageShouldFireQuotaUpdateEvent:783->lambda$addingMessageShouldFireQuotaUpdateEvent$18:783 
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] Expecting:
[1193686c0d88e18f827fb672dcd3a8f7150ed1be]  <QuotaSizeUsage{value=0}/QuotaSizeLimit{value=unlimited}>
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] to be equal to:
[1193686c0d88e18f827fb672dcd3a8f7150ed1be]  <QuotaSizeUsage{value=85}/QuotaSizeLimit{value=unlimited}>
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] but was not.
```

**Comment by Arsnael** [comment]
```
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] [ERROR]   MailboxManagerTest$EventTests.addingMessageShouldFireQuotaUpdateEvent:783->lambda$addingMessageShouldFireQuotaUpdateEvent$18:783 
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] Expecting:
[1193686c0d88e18f827fb672dcd3a8f7150ed1be]  <QuotaSizeUsage{value=0}/QuotaSizeLimit{value=unlimited}>
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] to be equal to:
[1193686c0d88e18f827fb672dcd3a8f7150ed1be]  <QuotaSizeUsage{value=85}/QuotaSizeLimit{value=unlimited}>
[1193686c0d88e18f827fb672dcd3a8f7150ed1be] but was not.
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**